python/paddle/fluid/optimizer.py

```diff
@@ -24,6 +24,7 @@
 from regularizer import append_regularization_ops
 from clip import append_gradient_clip_ops, error_clip_callback
 from contextlib import contextmanager
+from distribute_transpiler import UnionFind
```

Comment:
`optimizer.py` should not depend on distribute_transpiler, either put `_get_lr_decay_ops` in the transpiler or put `UnionFind` in a single file. I'd prefer the first method, because we don't have to change current demo files then.

Reply:
> optimizer.py should not depend on distribute_transpiler

Thanks @typhoonzero ,I think it's a good point, and maybe we can pass `Optimizer` instance to `transpile` interface so that we can support regularization for future.

Reply:
We can consider that when moving regularizer and clipping to the server side.

Reply:
Done, moved to transpiler.